Cannot set ykeys and labels dynamically

Hello everyone,

I have a chart that loads the ykeys and labels dynamically. I have the following code:
```
  public chartOptions = {
    xkey: 'date',
        ykeys: [],
        labels: [],
        parseTime:false,
        behaveLikeLine: true,
        pointFillColors:['#FAFAFA'],
        pointStrokeColors: ['#CCCCCC'],
        lineColors:['rgba(151,187,205,0.7)'],
        fillOpacity: 0.3,
        resize: true,
  }

this.eventService.getSalesChart(r['id']).subscribe(res => {
        this.chartOptions.ykeys = res.keyValuesObj;
        this.chartOptions.labels = res.keyValuesObj;
        this.salesData = res.chartDataValuesObj;
      });
```

After loading the data, my chartOptions object looks like this:
```
{
behaveLikeLine: true
fillOpacity: 0.3
labels:  ["secondrelease", "testprintticket", "total"]
lineColors: ["rgba(151,187,205,0.7)"]
parseTime: false
pointFillColors: ["#FAFAFA"]
pointStrokeColors: ["#CCCCCC"]
resize: true
xkey: "date"
ykeys: ["secondrelease", "testprintticket", "total"]
}
```

and my data object looks like this:
```
[{date: "01.10.2018", secondrelease: 0, testprintticket: 2565, total: 2565},
{date: "02.10.2018", secondrelease: 950, testprintticket: 17503, total: 18453}]
```

Even though the above seem correct, I get the following error:
`raphael.js:5957 Error: <path> attribute d: Expected moveto path command ('M' or 'm'), "L1041.45,309.012…".`

However, if I set the xkeys and labels in the default object initialization it works, but when they are set dynamically i always get the above error. Any help appreciated.

Carol

I did use [ ] for data and for options in my html file. I even changed my code to match yours:

HTML
`<div mk-morris-js [options]="chart.options" [data]="chart.data" type="Area" style="height: 300px; width: 100%;"></div>
`

TS
```
  public chart: any = { options: this.newChartOptions };

  public get newChartOptions(){
    return JSON.parse(JSON.stringify({
    xkey: 'date',
        ykeys: [],
        labels: [],
        parseTime:false,
        behaveLikeLine: true,
        pointFillColors:['#FAFAFA'],
        pointStrokeColors: ['#CCCCCC'],
        lineColors:['rgba(151,187,205,0.7)'],
        fillOpacity: 0.3,
        resize: true
    }));
  }
[...]

this.eventService.getSalesChart(r['id']).subscribe(res => {
        
        this.chart = { options: {
          xkey: 'date',
              ykeys: res.keyValuesObj,
              labels: res.keyValuesObj,
              parseTime:false,
              behaveLikeLine: true,
              pointFillColors:['#FAFAFA'],
              pointStrokeColors: ['#CCCCCC'],
              lineColors:['rgba(151,187,205,0.7)'],
              fillOpacity: 0.3,
              resize: true
          }, data: res.chartDataValuesObj };
      });
```

When chartOptions is initalized with a **new** object and then replaced with another object, like in the example above, i get the error: 

`raphael.js:5957 Error: <path> attribute d: Expected moveto path command ('M' or 'm'), "L1041.45,309.012…".`

When chartOptions is initalized with a **null** object and then set dynamically, I get the error:

```
ERROR TypeError: Cannot read property 'match' of undefined
    at Object.Morris.parseDate (morris.js:598)
    at Area.<anonymous> (morris.js:231)
    at Area.Morris.Grid.Grid.setData (morris.js:280)
    at MorrisChartDirective.push../node_modules/angular-morris-js/esm5/angular-morris-js.js.MorrisChartDirective.ngOnChanges (angular-morris-js.js:28)
    at checkAndUpdateDirectiveInline (core.js:9246)
    at checkAndUpdateNodeInline (core.js:10514)
    at checkAndUpdateNode (core.js:10476)
    at debugCheckAndUpdateNode (core.js:11109)
    at debugCheckDirectivesFn (core.js:11069)
    at Object.eval [as updateDirectives] (EventOverviewComponent.html:88)
```

Also, the error seems to happen when i set the data.

--- a/src/morris-chart.directive.ts
+++ b/src/morris-chart.directive.ts
@@ -15,7 +15,7 @@ export class MorrisChartDirective implements OnInit, AfterViewInit, OnChanges, O
   @Input() options: ChartOptions;
   @Input() data: ChartDatas;
 
-  /**
+  /*
    * [constructor description]
    * @method constructor
    * @param  {ElementRef} private elementRef [description]
@@ -26,7 +26,7 @@ export class MorrisChartDirective implements OnInit, AfterViewInit, OnChanges, O
     private ngZone: NgZone
   ) {}
 
-  /**
+  /*
    * [ngOnInit description]
    * @method ngOnInit
    */
@@ -36,7 +36,7 @@ export class MorrisChartDirective implements OnInit, AfterViewInit, OnChanges, O
     this._options.data = this.data;
   }
 
-  /**
+  /*
    * [ngAfterViewInit description]
    * @method ngAfterViewInit
    */
@@ -50,18 +50,19 @@ export class MorrisChartDirective implements OnInit, AfterViewInit, OnChanges, O
     }
   }
 
-  /**
+  /*
    * [ngOnChanges description]
    * @method ngOnChanges
    * @param  {SimpleChanges} changes [description]
    */
   ngOnChanges(changes: SimpleChanges) {
-    if(changes.data && !changes.data.firstChange) {
+    if((changes.data && !changes.data.firstChange) || (changes.options && !changes.options.firstChange)) {
+      Object.assign(this.chartInstance.options, this.options);
       this.chartInstance.setData(this.data);
     }
   }
 
-  /**
+  /*
    * [ngOnDestroy description]
    * @method ngOnDestroy
    */

